tests/test.c

```diff
@@ -1990,6 +1990,44 @@ static void test_netrc(void)
 	rmdir(".test");
 }
 
+static void test_robots(void)
+{
+	ROBOTS *robots;
+	ROBOTS_PATH *path;
+	const char *sitemap;
+	const char *data =
```

Comment:
You should declare a test_data structure with input (what now is 'data') and with expected output (e.g. array of paths, array of sitemaps). For example, see one function further up - test_netrc().